src/back/TicketManager/tests/PaymentService/TicketManager.PaymentService.ServicesTests/Endpoints/Payments/CancelPaymentEndpointTest.cs

```diff
@@ -0,0 +1,77 @@
+using FastEndpoints;
+using FluentAssertions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.EntityFrameworkCore;
+using Moq;
+using TicketManager.PaymentService.Contracts.Payments;
+using TicketManager.PaymentService.Domain.Payments;
+using TicketManager.PaymentService.Services.DataAccess;
+using TicketManager.PaymentService.Services.DataAccess.Repositories;
+using TicketManager.PaymentService.Services.Endpoints.Payments;
+using Xunit;
+
+namespace TicketManager.PaymentService.ServicesTests.Endpoints.Payments;
+
+public class CancelPaymentEndpointTest
+{
+    private readonly CancelPaymentEndpoint endpoint;
+    private readonly Mock<Payment> paymentMock;
+    private readonly Payment payment;
+    private readonly Mock<Repository<Payment, Guid>> paymentsMock;
+    
+    public CancelPaymentEndpointTest()
+    {
+        paymentMock = new Mock<Payment>();
+        payment = paymentMock.Object;
+        var dbContextMock = new Mock<PaymentDbContext>(new DbContextOptionsBuilder<PaymentDbContext>().Options);
+        var dbContext = dbContextMock.Object;
+        paymentsMock = new Mock<Repository<Payment, Guid>>(dbContext);
+        paymentsMock.Setup(p => p.FindAndEnsureExistenceAsync(payment.Id, default)).ReturnsAsync(payment);
+        var payments = paymentsMock.Object;
+        endpoint = Factory.Create<CancelPaymentEndpoint>(payments);
+    }
+    
+    [Fact]
+    public async Task WhenItIsCalledAndIsSuccessful_ItShouldCancelPayment()
+    {
+        paymentMock.Setup(p => p.CancelPayment()).Returns(true);
```

Comment:
Domain object is not dependency service to mock it.

Reply:
Addressed in 7f75126c.